File: domain/src/main/java/no/nav/melosys/domain/mottatteopplysninger/data/Periode.java
package no.nav.melosys.domain.mottatteopplysninger.data;

import java.time.LocalDate;
import javax.xml.bind.annotation.XmlType;

import no.nav.melosys.domain.ErPeriode;

@XmlType(name = "Periode", namespace = "soeknad") // Løser navnekonflikt med no.nav.melosys.domain.dokument.felles.Periode
public class Periode implements ErPeriode {

    private LocalDate fom;

    private LocalDate tom;

    public Periode () {
    }

    public Periode (LocalDate fom, LocalDate tom) {
        this.fom = fom;
        this.tom = tom;
    }

    @Override
    public LocalDate getFom() {
        return fom;
    }

    @Override
    public LocalDate getTom() {
        return tom;
    }

    @Override
    public String toString() {
        return new StringBuilder().append(fom).append(" → ").append(tom).toString();
    }
}

File: domain/src/main/kotlin/no/nav/melosys/domain/mottatteopplysninger/data/Periode.kt
package no.nav.melosys.domain.mottatteopplysninger.data

import no.nav.melosys.domain.MuligPeriode
import java.time.LocalDate

class Periode(
    override var fom: LocalDate? = null,
    override var tom: LocalDate? = null
) : MuligPeriode {
    override fun toString(): String = "$fom → ${tom ?: "∞"}"
}
